### DISCUSSION TITLE
Value for %energy#power% not working in all rules

### DISCUSSION THREAD

**Discussion by kuhnino** [arendst/Tasmota#14417] [Q&A] [answered]
May I´m a bit "stupid,  but the term "engery#power" is working here,

- ON energy#power<390 Do BACKLOG var1 0; var2 0 ENDON

but using the same term here inside not

- ON Time#Minute DO websend [WASCHKUECHE-STATUS-TW] var1 %energy#power% ENDON

I believe it´s only an understanding problem, but I couldnt find the root cause of my weired thinking

**Comment by ascillato** [comment, marked as answer]
Hi,

The issue is just a syntax error. `energy#power` is a trigger not a variable you can call. The list of supported variables are in the docs.
Besides, var1 is a command to store data inside it. Can't be used like that.

To do what you want, you can:

(to store the value of power)
`ON energy#power Do var1 %value% ENDON`

(to send it)
`ON Time#Minute DO websend [IP] %var1% ENDON`

**Reply by kuhnino** [reply]
ok, solves half of my problem then.

Can I have access to the other power values of energy#power also then in some way?

Like values from any other sensor??? 
![2022-01-13 15_42_32-Warmwasser-Strom - Konsole — Mozilla Firefox](https://user-images.githubusercontent.com/77860530/149351474-142e5b78-b248-43e3-920b-054d75493bb0.png)

**Reply by ascillato** [reply]
>Can I have access to the other power values of energy#power also then in some way?

yes. Everything under ENERGY can be used as a trigger like:
ENERGY#TOTAL
ENERGY#TODAY
in the same way as before
ON energy#total Do var2 %value% ENDON

Besides that, you can join both of your rules to do the websend in the same one that the trigger:

```js
Rule1
ON Energy#Power DO Websend [IP] %value% ENDON
ON Energy#Total DO Websend [IP] %value% ENDON
ON Energy#Today DO Websend [IP] %value% ENDON
```

Anyway, why not using MQTT? it will be populated automatically.

**Reply by kuhnino** [reply]
good question about  MQTT, want to protect myself from more "homewrk" maybe ????

Anyway, your help is really much apreciated, ticket is answered. I will close or mark it as solved .

**Reply by sfromis** [reply]
MQTT is excellent to *reduce* the amount of "homework" needed :-)
